extension/scripts/Selector/SelectorTable.js

```diff
@@ -158,19 +164,69 @@ var SelectorTable = {
         });
         return dataColumns;
     },
-
-    getFeatures: function () {
-        return ['selector', 'multiple', 'columns', 'delay', 'tableDataRowSelector', 'tableHeaderRowSelector', 'tableAddMissingColumns']
+    isExtractedDataColumns: function (header) {
+        answer = undefined;
+        this.columns.forEach(function (column) {
+            if (column.extract === true && header === column.header.trim()) {
+                answer = column.name;
+            }
+        });
+        return answer;
     },
 
-    getItemCSSSelector: function () {
-        return "table";
+
+
+    /**
+     * Extract table header column info from html
+     * @param html
+     */
+    getTableHeaderColumnsFromHTML: function (headerRowSelector, html, verticalTable) {
+        var $table = $(html);
+        var $headerRowColumns = $table.find(headerRowSelector);
+
+        if ($headerRowColumns.length > 1) {
+            $headerRowColumns = $headerRowColumns.find("th:first-child")
+        } else if ($headerRowColumns.find("th").length) {
+            $headerRowColumns = $headerRowColumns.find("th");
+        } else if ($headerRowColumns.find("td").length) {
+            $headerRowColumns = $headerRowColumns.find("td");
+        }
+        var columns = [];
+        if (verticalTable){
+            $headerRowColumns.prevObject.each(function (i, columnEl) {
```

Comment:
- code blocks under if-else alternatives are almost identical, you can do with one
- `if (header.length !== 0)` probably may be replaced with `if (header)`